fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/RuleSet.java

```diff
@@ -263,11 +264,18 @@ public class RuleSet {
                 new PushDownApplyLeftRule()
         ));
 
-        REWRITE_RULES.put(RuleSetType.SUBQUERY_REWRITE, ImmutableList.of(
+        REWRITE_RULES.put(RuleSetType.SUBQUERY_REWRITE_STAGE_1, ImmutableList.of(
```

Comment:
why do we need to split the sub query rewrite to three stages?  what's the standard？would better use a concrete name for stage 1, 2,3.

Reply:
Here is the reason why I split one rule group into three. For the new rule `ScalarApply2AnalyticRule`, it has more complex pattern than the other `Apply2xxxRule`, so, if we put them together in one rule set, the former one may lose the chance to do the transformation.

So I need to let `ScalarApply2AnalyticRule` be performed earlier than the others.

As for the naming, I haven't figure out a better name yet.